src/Microsoft.AspNetCore.ResponseCompression/ResponseCompressionProvider.cs

```diff
@@ -126,8 +136,10 @@ public virtual bool ShouldCompressResponse(HttpContext context)
                 mimeType = mimeType.Trim();
             }
 
-            // TODO PERF: StringSegments?
-            return _mimeTypes.Contains(mimeType);
+            return ShouldCompressExact(mimeType) //check exact match type/subtype
+                ?? ShouldCompressPartial(mimeType) //check partial match type/*
+                ?? ShouldCompressExact("*/*") //check wildcard */*
```

Comment:
Can exclude `*/*` ever override anything except an include `*/*`? Not sure why someone would define both. This could be simplified to only check for include `*/*`.